fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/TrinoFnCallTransformers.java

```diff
@@ -0,0 +1,136 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.parser.trino;
+
+import org.apache.doris.nereids.analyzer.PlaceholderExpression;
+import org.apache.doris.nereids.analyzer.UnboundFunction;
+import org.apache.doris.nereids.parser.ParserContext;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.functions.Function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The builder and factory for {@link org.apache.doris.nereids.parser.trino.TrinoFnCallTransformer},
+ * and supply transform facade ability.
+ */
+public class TrinoFnCallTransformers {
+
+    private static final ImmutableListMultimap<String, AbstractFnCallTransformer> TRANSFORMER_MAP;
+    private static final ImmutableListMultimap<String, AbstractFnCallTransformer> COMPLEX_TRANSFORMER_MAP;
+    private static final ImmutableListMultimap.Builder<String, AbstractFnCallTransformer> transformerBuilder =
+            ImmutableListMultimap.builder();
+    private static final ImmutableListMultimap.Builder<String, AbstractFnCallTransformer> complexTransformerBuilder =
+            ImmutableListMultimap.builder();
+
+    static {
+        registerTransformers();
+        TRANSFORMER_MAP = transformerBuilder.build();
+        registerComplexTransformers();
+        COMPLEX_TRANSFORMER_MAP = complexTransformerBuilder.build();
+    }
+
+    private TrinoFnCallTransformers() {
+    }
+
+    /**
+     * Function transform facade
+     */
+    public static Function transform(String sourceFnName, List<Expression> sourceFnTransformedArguments,
+            ParserContext context) {
+        List<AbstractFnCallTransformer> transformers = getTransformers(sourceFnName, false);
+        Function function = doTransform(transformers, sourceFnName, sourceFnTransformedArguments, context);
+        if (function != null) {
+            return function;
+        }
+        transformers = getTransformers(sourceFnName, true);
```

Comment:
maybe we should return complex == false and == true togather, and let complex == false ahead of complex == true

Reply:
Good idea, have fixed it